mango-core/src/main/java/org/calrissian/mango/domain/AttributeStore.java

```diff
@@ -47,6 +37,9 @@
      */
     <T> Attribute<T> get(String key);
 
+    <T> Attribute<T> get(String key, Class<T> clazz);
```

Comment:
When a user knows something should be single-valued, they should just be able to ask for one (which happens to be the first to come back). Now that the events serialized into Accumulo values are fairly idempotent, there's less chance of someone accidentally adding an attribute later and turning a single-value into a mutli-value.